src/src/test/java/StringManipulationTest.java

```diff
@@ -0,0 +1,183 @@
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class StringManipulationTest {
+
+    private StringManipulationInterface manipulatedstring;
+
+    @BeforeEach
+    public void setUp() {
+        manipulatedstring = new StringManipulation();
+    }
+
+    @AfterEach
+    public void tearDown() {
+        manipulatedstring = null;
+    }
+
+    @Test
+    public void testCount1() {
+        manipulatedstring.setString("This is my string");
+        int length = manipulatedstring.count();
+        assertEquals(4, length);
+    }
+
+    @Test
+    public void testCount2() {
+        manipulatedstring.setString("One more string for testing");
+        int length = manipulatedstring.count();
+        assertEquals(5, length);
+    }
+
+    @Test
+    public void testCount3() {
+        manipulatedstring.setString("");
+        int length = manipulatedstring.count();
+        assertEquals(0, length);
+    }
+
+    @Test
+    public void testCount4() {
+        manipulatedstring.setString("SingleWord");
+        int length = manipulatedstring.count();
+        assertEquals(1, length);
+    }
+
+    @Test
+    public void testRemoveNthCharacter1() {
+        manipulatedstring.setString("I'd b3tt3r put s0me d161ts in this 5tr1n6, right?");
+        assertEquals("I' bttr uts0e 16tsinths trn6 rgh?", manipulatedstring.removeNthCharacter(3, false));
+    }
+
+    @Test
+    public void testRemoveNthCharacter2() {
+        manipulatedstring.setString("I'd b3tt3r put s0me d161ts in this 5tr1n6, right?");
+        assertEquals("I'  b tt r  ut s0 e  16 ts in th s  tr n6  r gh ?", manipulatedstring.removeNthCharacter(3, true));
+    }
+
+    @Test
+    public void testRemoveNthCharacter3() {
+        manipulatedstring.setString("");
+        assertEquals("", manipulatedstring.removeNthCharacter(2, false));
+    }
+
+    @Test
+    public void testRemoveNthCharacter4() {
+        manipulatedstring.setString("111111111");
+        assertEquals("1 1 1 1 1", manipulatedstring.removeNthCharacter(2, true));
+    }
+
+    @Test
+    public void testRemoveNthCharacter5() {
+        manipulatedstring.setString("111111111");
+        assertEquals("11111", manipulatedstring.removeNthCharacter(2, false));
+    }
+
+    @Test
+    public void testRemoveNthCharacter6() {
+        manipulatedstring.setString("1");
+        assertEquals("1", manipulatedstring.removeNthCharacter(4, true));
+    }
+
+    @Test
+    public void testRemoveNthCharacter7() {
+        manipulatedstring.setString("123");
+        assertEquals("12", manipulatedstring.removeNthCharacter(3, false));
+    }
+
```

Comment:
You handle the testSubString cases really well but you may want to add exception handling for if the indices are out of bound.